### PR TITLE
config: only allow confidential instances on stackit

### PR DESCRIPTION
**Pull request by h3adex**
### Context
When creating a confidential Kubernetes cluster on Stackit, customers have occasionally confused confidential flavor IDs with non-confidential ones which resulted in a "stuck" creation period.

### Proposed Change(s)
I have implemented an instance type check for the Stackit cloud provider to prevent this confusion.

### Related Issue
[#3438](https://github.com/edgelesssys/constellation/issues/3438)

### Additional Info
I have rebuilt the Constellation CLI and confirmed that the change works correctly on a Linux x86 machine.
Using ```bazel build //cli:cli_oss```

### Checklist
- [ ] Run the E2E tests that are relevant to this PR's changes

### CODE REVIEW

**Review by msanft — APPROVED** [review]
LGTM. Also thanks a lot for adding a test!

File: internal/config/validation.go [hunk 1]
```diff
@@ -549,6 +549,16 @@ func validInstanceTypeForProvider(insType string, attestation variant.Variant) b
 			}
 		}
 	case variant.QEMUVTPM{}, variant.QEMUTDX{}:
+		// only allow confidential instances on stackit cloud using QEMU vTPM
+		if cloud := provider.OpenStack.Cloud; strings.ToLower(cloud) == "stackit" {
```
Comment (msanft) [inline internal/config/validation.go:553]: We need an `if provider.OpenStack != nil` here, as there's more scenarios where we'll have `case variant.QEMUVTPM{}, variant.QEMUTDX{}`, such as MiniConstellation, and else we'd run into a nil panic there here when dereferencing `provider.OpenStack`.

We might also want "success" test case for these variants without the OpenStack / STACKIT case here.

**Review by msanft — CHANGES_REQUESTED** [review]

**Comment by msanft** [comment]
The failing CI is somewhat expected on your branch, as the required tokens aren't available there. I'll verify that these succeed on my machine once we're ready to merge, so don't worry about fixing that for now. :)

File: internal/config/validation.go (hunk 1 above)
Comment (h3adex) [reply internal/config/validation.go]: I've added a provider check and two test cases. Although I'm not entirely sure which provider Miniconstellation is using, I made an educated guess: `ProviderConfig{QEMU: &QEMUConfig{}}` with the variant `variant.QEMUTDX{}`.

**Review by daniel-weisse — APPROVED** [review]

**Comment by msanft** [comment]
All tests succeed, merging.

**Pull request merged**